R/build-favicons.R

```diff
@@ -78,34 +87,46 @@ build_favicon <- function(pkg = ".") {
   tmp <- tempfile()
   on.exit(unlink(tmp))
   result <- httr::GET(result$favicon$package_url, httr::write_disk(tmp))
-  utils::unzip(tmp, exdir = path(pkg$src_path, "pkgdown", "favicon"))
+
+  tryCatch({
+    utils::unzip(tmp, exdir = path(pkg$src_path, "pkgdown", "favicon"))
+  },
+  warning = function(e) {
+    warning("Your logo file couldn't be processed and may be corrupt.")
+  },
+  error = function(e) {
+    warning("Your logo file couldn't be processed and may be corrupt.")
```

Comment:
It still feels weird to turn this into a warning; especially since it loses all context so it will be hard to figure out what's gone wrong.